src/ucp/wireup/wireup_cm.c

```diff
@@ -749,10 +749,12 @@ static unsigned ucp_ep_cm_disconnect_progress(void *arg)
          * anymore */
         goto out;
     } else if (ucp_ep->flags & UCP_EP_FLAG_CLOSED) {
-        /* if an EP was closed and not local connected anymore, not failed
-         * and no CLOSE request is set, it means that an EP was disconnected
-         * from a peer */
-        ucs_assert(ucp_ep->flags & UCP_EP_FLAG_DISCONNECTED_CM_LANE);
+        /* if an EP was closed and not local connected anymore (i.e.
+         * ucp_ep_cm_disconnect_cm_lane() was called from ucp_ep_close_nbx()),
+         * not failed and no CLOSE request is set, it means that an EP was
+         * disconnected from a peer */
+        ucs_assert((ucp_ep->flags & UCP_EP_FLAG_DISCONNECTED_CM_LANE) &&
+                   !(ucp_ep->flags & UCP_EP_FLAG_ERR_HANDLER_INVOKED));
```

Comment:
minor: pls break up assertion for each statement separately

Reply:
fixed